lib/engine/step/g_18_tn/buy_company.rb

```diff
@@ -20,6 +20,18 @@ def can_buy_company?(entity)
             companies.any? &&
             companies.map(&:value).min <= entity.cash
         end
+
+        private
+
+        def already_bought_in_turn_one?(entity)
+          return false if @game.turn != 1 || !entity.corporation?
+
+          # A corporation may buy at most one company during turn 1
+          return true if entity.companies.any?
+
+          # A player may sell at most one company during turn 1
+          @round.company_sellers.include?(entity.owner)
```

Comment:
Is a corporation allowed to buy a company from a player other than the corporation's president? I'll test this real quick, but I think this line would break that. (though I'm sure even if allowed, this would be a rare case...)

Reply:
Confirmed: michaeljb3 is president of L&N, which buys Memphis & Charleston from michaeljb2. Then michaeljb2 is president of SR, which does not have the option to buy a company from anyone, though it should be able to buy from a player other than michaeljb2.

https://gist.github.com/michaeljb/1e72c407aa61102a67f66f51e6486785

Reply:
Another contrived example where Player 2 gets two presidencies in the first round, the first corporation buys a company from Player 2, and then the second corporation doesn't have the option to buy the private from any other players - https://gist.github.com/michaeljb/db071715e6c6d9fcfc364f7ad8721965

Reply:
Good catch.

In 18TN the following is allowed in first turn:
1. Players can buy from other players during SR1
2. Corporations can buy at most one private
3. Players can sell at most one private to a corporation

Have now improved the code so it will just stop buy of company if it is breaking the rules of OR1. So it should not affect SR1 or later ORs.

Reply:
With e10fd174 checked out, in both of the hotseat games I linked in my prior comments, SR is still unable to buy a company in OR1. L&N bought a company from the SR's president, but SR should still be allowed to buy from another player.

I think the logic for corporation buying one and player selling one need to be separated in order to handle this edge case correctly.

Reply:
@michaeljb Try again.
SR buy not available was a major bug in 18TN.